### PR TITLE
pin binary dependency versions

### PR DESCRIPTION
**Pull request by sadlerap**
This lets us track binary dependencies through dependabot without having to maintain a bin/ folder of magic binaries.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- package-ecosystem: github-actions
+  directory: "/"
+  schedule:
+    interval: weekly
+  open-pull-requests-limit: 10
+- package-ecosystem: gomod
+  directory: "/server"
+  schedule:
+    interval: weekly
+  open-pull-requests-limit: 10
```
Comment (filariow) [inline .github/dependabot.yml:12]: We also need to monitor the `/operator` folder

**Review by filariow — APPROVED** [review]
Nice, just a small nit.

File: .github/dependabot.yml (hunk 1 above)
Comment (sadlerap) [reply .github/dependabot.yml:12]: Fixed in `abbcc0c`.

**Comment by sadlerap** [comment]
I'm going to leave converting the tools under `/operator` to use these for a separate PR once we start working on stuff there.

**Review by filariow — DISMISSED** [review]
Mostly LGTM. 

Maybe moving them under `hack/tools` instead of just `hack`?

**Review by filariow — APPROVED** [review]
/lgtm

**Pull request merged**